Formula/wp-cli.rb

```diff
@@ -17,6 +17,8 @@ class WpCli < Formula
     sha256 cellar: :any_skip_relocation, mojave:        "67acd9816806eef402f59f1904fcebd2e23e6d6cb7657604430e299cf21bd300"
   end
 
+  uses_from_macos "php"
```

Comment:
Why not use `on_linux` for linux build dep?

Reply:
This should be a runtime dependency, which has been passing on macOS due to default PHP (which I guess will fail in macOS 12.0 Monterey since PHP is no longer included).

Mainly based on how other Formulae handle this (e.g. `php-cs-fixer` and `phpbrew`). Other Formulae that have PHP dependency use Homebrew's version for both macOS and Linux.